### DISCUSSION TITLE
Task Instance Queue Not Preserved on Clear

### DISCUSSION THREAD

**Discussion by JGoldman110** [apache/airflow#25320] [General]
Currently we have a `task_instance_mutation_hook` which dynamically changes the queue a task instance is assigned to based on if the task instance is a scheduled or manual run. This works well, except for when a user clears a task instance. 

When clearing a task instance the `task_instance_mutation_hook` is not called.  Instead when clearing a task instance the task instance is refreshed and the original queue assigned to that task instance in the `task_instance_mutation_hook` is overridden and set to the default queue defined in the operator or the default queue set in airflow configs.

Does it make sense to also call the `task_instance_mutation_hook` on clear as well? or to preserve the original queue on the task instance on clear? Or maybe there is another hook to modify task_instance queue on clear that I am not aware of?

**Comment by JGoldman110** [comment]
@milton0825 any thoughts on this?

**Comment by potiuk** [comment]
There is already an issue about it. https://github.com/apache/airflow/issues/20143 . You can chime in there.

**Reply by JGoldman110** [reply]
Thanks @potiuk will chime in over there
